### PR TITLE
fix(k8s): consistent types in manifest.Manifest

### LINKED ISSUE #318: After Upgrade to v0.11.0 tk apply shows diff on Ingress annotations
Hi,

v0.11.0 shows a diff in annotation of Ingress objects, even after I've deleted the ingress and recreate it with v0.11.0.
Going back to v0.10.0 solves the problem immediately.

Does anybody else see this kind of problem?

In fact tanka wants to remove ` kubectl.kubernetes.io/last-applied-configuration:` from the Ingress objects.

Best regards and thanks
Andreas

### PR DESCRIPTION
**Pull request by sh0rez**
The manifest.Manifest is a `map[string]interface{}`, which sticks to what `json.Unmarshal` returns, meaning it only includes concrete values, `[]interface{}` and `map[string]interface{}`.

In a v0.10 change I accidentally broke that, without realizing it: https://github.com/grafana/tanka/blob/a6d5fde5aac69dc4d66bb57d5a501e52ba60fada/pkg/kubernetes/manifest/manifest.go#L248

Up to #312, this didn't create any issues. Now however, because we called `Annotations` on all objects, types weren't like they should have been.

This caused subsetdiff to miss annotations, not computing the subset on them at all.

Fixes #318

### CODE REVIEW

File: pkg/kubernetes/manifest/manifest.go [hunk 1]
```diff
@@ -219,37 +219,22 @@ func (m Metadata) UID() string {
 }
 
 // Labels of the manifest
-func (m Metadata) Labels() map[string]string {
-	return safeStringMap(m, "labels")
+func (m Metadata) Labels() map[string]interface{} {
+	return safeMSI(m, "annotations")
```
Comment (malcolmholmes) [inline pkg/kubernetes/manifest/manifest.go:223]: Erm shouldn't this be labels?

File: pkg/kubernetes/subsetdiff.go [hunk 2]
```diff
@@ -93,60 +92,55 @@ func subsetDiff(c client.Client, m manifest.Manifest) (*difference, error) {
...
-	}
+	should := m.String()
 
-	is, err := yaml.Marshal(subset(m, rawIs))
-	if err != nil {
-		return nil, err
-	}
-	if string(is) == "{}\n" {
-		is = []byte("")
+	sub := subset(m, rawIs)
```
Comment (malcolmholmes) [inline pkg/kubernetes/subsetdiff.go:97]: Where is rawIs set? From what I can see it is only an empty MSI here.

File: pkg/kubernetes/manifest/manifest.go (hunk 1 above)
Comment (sh0rez) [reply pkg/kubernetes/manifest/manifest.go:223]: Oh shit, yeah

File: pkg/kubernetes/subsetdiff.go (hunk 2 above)
Comment (sh0rez) [reply pkg/kubernetes/subsetdiff.go:97]: https://github.com/grafana/tanka/blob/2ee719f219079a91c8f03a39bc2a17399eab2874/pkg/kubernetes/subsetdiff.go#L82-L87

**Pull request merged**